theano/gof/op.py

```diff
@@ -783,6 +783,9 @@ def __new__(cls, *args, **kwargs):
         # this function exists to silently and transparently ensure that all
         # existing Ops get a _op_use_c_code attribute
         obj = object.__new__(cls)
+        # Ensuring all the Ops created henceforth has __props__
+        if not hasattr(obj, '__props__'):
+            raise
```

Comment:
Also, by using props, we can cache without building in all the parameters from `*args` and `**kwargs` and rebuilding those parameters to initialize the new class instance. (In one comment you had asked me to use `__props__` for this purpose.